File: src/main/java/um/haberes/core/hexagonal/liquidaciones/categoria/domain/model/Categoria.java
package um.haberes.core.hexagonal.liquidaciones.categoria.domain.model;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Categoria {

    private Integer categoriaId;

    @Builder.Default
    private String nombre = "";

    @Builder.Default
    private BigDecimal basico = BigDecimal.ZERO;

    @Builder.Default
    private Byte docente = 0;

    @Builder.Default
    private Byte noDocente = 0;

    @Builder.Default
    private Byte liquidaPorHora = 0;

    @Builder.Default
    private BigDecimal estadoDocente = BigDecimal.ZERO;
}

File: src/main/java/um/haberes/core/kotlin/model/Categoria.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal

@Entity
@Table
data class Categoria(

    @Id
    var categoriaId: Int? = null,

    var nombre: String = "",
    var basico: BigDecimal = BigDecimal.ZERO,
    var docente: Byte = 0,
    var noDocente: Byte = 0,
    var liquidaPorHora: Byte = 0,
    var estadoDocente: BigDecimal = BigDecimal.ZERO

) : Auditable()
